rollouts/api/v1alpha1/rollout_types.go

```diff
@@ -107,17 +107,21 @@ type ClusterSourceKind struct {
 const (
 	GitHub PackageSourceType = "GitHub"
 	GitLab PackageSourceType = "GitLab"
+	OCI    PackageSourceType = "OCI"
 )
 
-// +kubebuilder:validation:Enum=GitHub;GitLab
+// +kubebuilder:validation:Enum=GitHub;GitLab;OCI
 type PackageSourceType string
 
 // PackagesConfig defines the packages the Rollout should deploy.
 type PackagesConfig struct {
 	SourceType PackageSourceType `json:"sourceType"`
 
-	GitHub GitHubSource `json:"github,omitempty"`
-	GitLab GitLabSource `json:"gitlab,omitempty"`
+	// TODO(droot): Change Github and Gitlab to pointers because
```

Comment:
As long as we are using a discriminator here, it probably doesn't matter that much if they are pointers or not. Nil objects should be fine, since we will be using the discriminator value to determine which one to use rather than check for nil.

Reply:
You are right. Though, `nil` does help with the serialization of `omitempty` semantics.